WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift

```diff
@@ -29,17 +30,20 @@ final class PointOfSaleDashboardViewModel: ObservableObject {
     @Published var isEmpty: Bool = false
     @Published var showExitPOSModal: Bool = false
     @Published var showSupport: Bool = false
+    @Published var showsConnectivityError: Bool = false
 
     private var cancellables: Set<AnyCancellable> = []
 
     init(cardPresentPaymentService: CardPresentPaymentFacade,
          totalsViewModel: any TotalsViewModelProtocol,
          cartViewModel: any CartViewModelProtocol,
-         itemListViewModel: any ItemListViewModelProtocol) {
+         itemListViewModel: any ItemListViewModelProtocol,
+         connectivityObserver: ConnectivityObserver = ServiceLocator.connectivityObserver) {
```

Comment:
Do we want to have a reference to the ServiceLocator here? I'm thinking it should stay outside the POS viewmodels. I understand that if we do not provide it, then this forces us to also declare it on existing previews, or make it optional which it's not idea either 🤔